packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/useUploadView.ts

```diff
@@ -49,41 +48,54 @@ export const useUploadView = (params?: {
   const onActionStart = React.useCallback(() => {
     handleProcess({
       config: getInput(),
-      destinationPrefix,
+      destinationPrefix: key,
       options: { preventOverwrite: !isOverwriteEnabled },
     });
-  }, [destinationPrefix, getInput, handleProcess, isOverwriteEnabled]);
+  }, [isOverwriteEnabled, key, getInput, handleProcess]);
 
   const onActionCancel = React.useCallback(() => {
     tasks.forEach((task) => task.cancel?.());
   }, [tasks]);
 
   const onExit = React.useCallback(() => {
-    if (isFunction(_onExit)) _onExit?.(current!);
     // clear tasks state
     tasks.forEach(({ remove }) => remove());
     // clear files state
     dispatchStoreAction({ type: 'RESET_FILE_ITEMS' });
     // clear selected action
     dispatchStoreAction({ type: 'RESET_ACTION_TYPE' });
+    _onExit?.(current);
   }, [tasks, dispatchStoreAction, _onExit, current]);
 
   const onToggleOverwrite = React.useCallback(() => {
     setOverwriteEnabled((prev) => !prev);
   }, []);
 
+  const onTaskCancel = React.useCallback(
+    (task: Task) => {
+      if (isProcessing) {
+        task.cancel();
+      } else {
+        dispatchStoreAction({ type: 'REMOVE_FILE_ITEM', id: task.data.id });
+        task.remove();
+      }
+    },
+    [isProcessing, dispatchStoreAction]
+  );
```

Comment:
Once https://github.com/aws-amplify/amplify-ui/pull/6041 is merged the store action piece here can be provided as `onTaskRemove` and we can create these conditional callbacks for the Table directly in the view logic layer. 

Added a related note [here](https://github.com/aws-amplify/amplify-ui/pull/6041/files#r1835500404), due to the amount of logic we have been adding around conditional remove/cancel think we should drop the `remove` handler altogether and allow a `remove?: boolean` flag to `cancel`